ui/packages/components/src/RunDetailsV2/RunDetails.tsx

```diff
@@ -42,54 +41,29 @@ type Run = {
 export function RunDetails(props: Props) {
   const { getResult, getRun, getTrigger, pathCreator, pollInterval, rerun, runID, standalone } =
     props;
-  const [error, setError] = useState<Error>();
 
-  const [run, setRun] = useState<Run>();
-  const endedAt = run?.trace?.endedAt;
-  useEffect(() => {
-    if (endedAt) {
-      // Don't poll if the run has ended
-      return;
-    }
-
-    if (!pollInterval) {
-      if (run) {
-        // Nothing left to fetch
-        return;
+  const runRes = useQuery({
+    queryKey: ['run', runID],
+    queryFn: useCallback(() => {
+      return getRun(runID);
+    }, [getRun, runID]),
+    retry: 3,
+    refetchInterval: pollInterval,
```

Comment:
**idea**: Refetch interval allows for a callback function. We could make this dynamic so that it refetches until the job has a finalized status like completed or canceled (using `endedAt`) similar to the previous code.